assets/src/edit-story/output/story.js

```diff
@@ -0,0 +1,177 @@
+  return (
+    <>
+      <style
+        amp-boilerplate=""
+        dangerouslySetInnerHTML={{
+          __html:
+            'body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}',
+        }}
+      />
+      <noscript>
+        <style
+          amp-boilerplate=""
+          dangerouslySetInnerHTML={{
+            __html:
+              'body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}',
+          }}
+        />
+      </noscript>
+    </>
+  );
+}
+
+function CustomCSS() {
+  return (
+    <style
+      amp-custom=""
+      dangerouslySetInnerHTML={{
+        __html: `
+              .page-background-area, .page-safe-area {
+                position: absolute;
+                overflow: hidden;
+                margin: auto;
+                top: 0;
+                bottom: 0;
+                left: 0;
+                right: 0;
+              }
+
+              .page-background-area img, .page-background-area video {
+                object-fit: cover;
+              }
+
+              .wrapper {
+                position: absolute;
+                overflow: hidden;
+              }
+
+              .fill {
+                position: absolute;
+                top: 0;
+                left: 0;
+                right: 0;
+                bottom: 0;
+                margin: 0;
+              }
+              `,
+      }}
+    />
+  );
+}
+
+function OutputStory({ story, pages, metadata: { publisher, poster } }) {
+  const ampExtensions = getUsedAmpExtensions(pages);
+  return (
+    <html amp="" lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta
+          name="viewport"
+          content="width=device-width,minimum-scale=1,initial-scale=1"
+        />
+        {ampExtensions.map(({ name, src }) => (
+          <script key={src} async="" src={src} custom-element={name} />
+        ))}
+        <Boilerplate />
+        {/* Everything between these markers can be replaced server-side. */}
+        <meta name="web-stories-replace-head-start" />
+        <link rel="canonical" href={story.link} />
+        <CustomCSS />
+        <meta name="web-stories-replace-head-end" />
+      </head>
+      <body>
+        <amp-story
+          standalone="standalone"
+          publisher={publisher.name}
+          publisher-logo-src={publisher.logo}
+          title={story.title}
+          poster-portrait-src={story.featuredMediaUrl || poster}
```

Comment:
This won't be the correct image size. Not too dramatic, as it's still valid AMP, but perhaps we could somehow get the right one. That would require using the MediaProvider though.

@spacedmonkey Thoughts?

Reply:
Should be simple enough to add another field to the REST API. I already added the feature image in at medium size [here](https://github.com/google/web-stories-wp/blob/master/includes/REST_API/Stories_Controller.php#L90-L93). We could change this to return an object, key => value of size => url. Might be the cleanest way. I would prefer not to do a call to the rest API here. As it takes time.